linked_list/linked_list.py

```diff
@@ -86,10 +150,22 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def reverse(self):

```

Comment:
👍 How is the space complexity O(n)?  What secondary set of nodes are you creating?